integration-tests/15_achievements.ts

```diff
@@ -0,0 +1,1043 @@
+
+    const match = await prisma.match.findFirst({
+        where: {
+            pupilId: pupil.pupil.id,
+            dissolved: false,
+        },
+        select: { id: true },
+    });
+    await client.request(`
+        mutation PupilJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+    const pupil_joined_match_meeting_achievements = await prisma.user_achievement.findMany({
+        where: {
+            group: 'pupil_conduct_match_appointment',
+            userId: user.userID,
+        },
+    });
+    assert.ok(pupil_joined_match_meeting_achievements);
+    assert.notStrictEqual(pupil_joined_match_meeting_achievements.length, 0);
+});
+
+void test('Reward student regular learning', async () => {
+    await adminClient.request(`mutation ResetRateLimits { _resetRateLimits }`);
+    await createStudentRegularLearningTemplate();
+
+    const { student, client } = await studentOne;
+    const user = await getUser(student.userID);
+
+    const match = await prisma.match.findFirst({
+        where: {
+            studentId: student.student.id,
+            dissolved: false,
+        },
+        select: { id: true },
+    });
+    // request to generate the achievement with initial record value 1
+    await client.request(`
+        mutation StudentJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const date = new Date();
+    date.setDate(date.getDate() - 7);
+    await prisma.achievement_event.create({
+        data: {
+            userId: user.userID,
+            metric: 'student_match_learned_regular',
+            value: 1,
+            createdAt: date,
+            action: 'student_joined_match_meeting',
+            relation: `match/${match.id}`,
+        },
+    });
+    // request to set the achievements record value to 2 due to the past event generated
+    await client.request(`
+        mutation StudentJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const student_match_regular_learning_record = await prisma.user_achievement.findFirst({
+        where: {
+            userId: user.userID,
+            group: 'student_match_regular_learning',
+            achievedAt: { not: null },
+            recordValue: 2,
+        },
+    });
+    assert.ok(student_match_regular_learning_record);
+
+    await prisma.achievement_event.deleteMany({
+        where: {
+            userId: user.userID,
+            metric: 'student_match_learned_regular',
+            relation: `match/${match.id}`,
+        },
+    });
+    await client.request(`
+        mutation StudentJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const student_match_regular_learning = await prisma.user_achievement.findFirst({
+        where: {
+            userId: user.userID,
+            group: 'student_match_regular_learning',
+            achievedAt: null,
+            recordValue: 2,
+        },
+    });
+    assert.ok(student_match_regular_learning);
```

Comment:
What case is tested here?

Reply:
It's check, if this achievement is set to achievedAt = null and still hast recordValue 2, when the current streak does not have the highest number of events linked to the achievement. 

- First it is to be seen if all events included in the evaluation generate the record Value 2.
- The first event is deleted and the match meeting join ist executed to trigger the achievement evaluation again.
- It's been checked if the current achievement does not have the highest return value in the evaluation.

Reply:
Could you add this explanation as a comment?